hivelvet-frontend/src/App.tsx

```diff
@@ -0,0 +1,97 @@
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Hivelvet; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect } from 'react';
+import { INSTALLER_FEATURE } from './constants';
+import { webRoutes } from './routing/config';
+import { installRoutes } from './routing/config-install';
+
+import Router from './routing/Router';
+
+import './App.less';
+import { Layout, ConfigProvider, BackTop, Button } from 'antd';
+import { CaretUpOutlined } from '@ant-design/icons';
+
+import AppHeader from './components/AppHeader';
+import AppFooter from './components/AppFooter';
+import AppSider from './components/AppSider';
+
+import 'moment/locale/fr';
+import 'moment/locale/ar';
+import 'moment/locale/en-au';
+
+import Logger from './lib/logger';
+
+import authService from './services/auth.service';
+import LocaleService from './services/locale.service';
+import { withTranslation } from 'react-i18next';
+
+const { Content } = Layout;
+
+type userType = {
+    username: string;
+    email: string;
+    role: string;
+};
+
+const App = () => {
+    const [currentUser, setCurrentUser] = React.useState<userType>(null);
+    const [isLogged, setIsLogged] = React.useState<boolean>(false);
+    const [language, setLanguage] = React.useState<string>(LocaleService.language);
+
+    const setUser = (user: userType, logged: boolean) => {
+        setCurrentUser(user);
+        setIsLogged(logged);
+    };
+    const setLang = (lang: string) => {
+        setLanguage(lang);
+        LocaleService.changeLocale(lang);
+    };
+    useEffect(() => {
+        if (INSTALLER_FEATURE) Logger.info('Initialisation Hivelvet Installer Application');
+        else Logger.info('Initialisation Hivelvet Webapp Application');
+        const user: userType = authService.getCurrentUser();
+        if (authService.getCurrentUser() != null) setUser(user, true);
+    }, []);
+
+    return (
+        <Layout className={LocaleService.direction == 'rtl' ? 'page-layout-content-rtl' : 'page-layout-content'}>
+            <ConfigProvider locale={LocaleService.antdlocale} direction={LocaleService.direction} componentSize="large">
+                {isLogged && !INSTALLER_FEATURE && <AppSider />}
+                <Layout className="page-layout-body">
+                    <AppHeader
+                        currentUser={currentUser}
+                        currentLocale={language}
+                        setLang={setLang}
+                        isLogged={isLogged}
+                        setUser={setUser}
+                    />
+                    <Content className="site-content">
+                        <Router routes={INSTALLER_FEATURE ? installRoutes : webRoutes} setUser={setUser} />
```

Comment:
INSTALLER_FEATURE is a variable that should be used to remove code while minimised. Hiding features with "true" or "false" is not what we are looking for.

Please check this https://webpack.js.org/plugins/define-plugin/